lib/client/mfa/fido.go

```diff
@@ -0,0 +1,208 @@
+package mfa
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/segmentio/aws-okta/lib/client/types"
+
+	log "github.com/sirupsen/logrus"
+
+	u2fhost "github.com/marshallbrekka/go-u2fhost"
+)
+
+const (
+	MaxOpenRetries = 10
+	RetryDelayMS   = 200 * time.Millisecond
+)
+
+var (
+	errNoDeviceFound = fmt.Errorf("no U2F devices found. device might not be plugged in")
+)
+
+// FIDODevice is implementation of MFADevice for SMS
+type FIDODevice struct {
+	userInput     Input
+	codeRequested bool
+	id            string
+}
+
+func (d *FIDODevice) SetId(id string) {
+	d.id = id
+}
+
+func (d *FIDODevice) GetId() string {
+	return d.id
+}
+
+// Supported will check if the mfa config can be used by this device
+func (d *FIDODevice) Supported(factor types.OktaUserAuthnFactor) error {
+	if factor.FactorType == "u2f" && factor.Provider == "FIDO" {
+		d.id = factor.Id
+		return nil
+	}
+	return fmt.Errorf("FIDODevice doesn't support %s %w", factor.FactorType, types.ErrNotSupported)
+}
+
+// Verify is called to get generate the payload that will be sent to Okta.
+//   We will call this twice, once to tell Okta to send the code then
```

Comment:
I'm not really clear on this flow. The first time through, `d.codeRequested` will be `false`, so it gets set to `true`, then we return a blank `PassCode`?

Reply:
I think the separation of concerns between an `mfa.Device` and its caller needs to be more clearly specified. The fact that you're passing on-the-wire structs between them to me indicates that the Device is more information than it needs

Reply:
I've updated the logic to make it rely on the response from Okta rather than how many times Verify is called. Hopefully this is a better example of how we might use this Device interface and addresses your concerns.